lib/compat/wordpress-6.0/register-webfonts-from-theme-json.php

```diff
@@ -122,28 +123,28 @@ function gutenberg_add_registered_webfonts_to_theme_json( $data ) {
 		$data['settings']['typography']['fontFamilies'] = array();
 	}
 
-	// Add missing fonts.
-	foreach ( $to_add as $family ) {
-		$font_face = array();
-		foreach ( $font_families_registered as $font_family ) {
-			if ( $family !== $font_family['font-family'] ) {
-				continue;
-			}
-			$camel_cased = array();
-			foreach ( $font_family as $key => $value ) {
+	foreach ( $to_add as $slug ) {
+		$font_faces_for_family = $font_families_registered[ $slug ];
+		$family_name           = $font_faces_for_family[0]['font-family'];
+		$font_faces            = array();
+
+		foreach ( $font_faces_for_family as $font_face ) {
+			$camel_cased = array( 'origin' => 'gutenberg_wp_webfonts_api' );
+			foreach ( $font_face as $key => $value ) {
 				$camel_cased[ lcfirst( str_replace( '-', '', ucwords( $key, '-' ) ) ) ] = $value;
 			}
-			$font_face[] = $camel_cased;
+			$font_faces[] = $camel_cased;
 		}
+
 		$data['settings']['typography']['fontFamilies'][] = array(
-			'fontFamily' => false !== strpos( $family, ' ' ) ? "'{$family}'" : $family,
-			'name'       => $family,
-			'slug'       => sanitize_title( $family ),
-			'fontFace'   => $font_face,
+			'fontFamily' => false !== strpos( $family_name, ' ' ) ? "'{$family_name}'" : $family_name,
+			'name'       => $family_name,
+			'slug'       => $slug,
+			'fontFace'   => $font_faces,
 		);
 	}
 
 	return $data;
 }
 
-add_action( 'wp_loaded', 'gutenberg_register_webfonts_from_theme_json' );
+add_action( 'after_setup_theme', 'gutenberg_register_webfonts_from_theme_json' );
```

Comment:
This is necessary so we can register the webfonts from `theme.json` as soon as possible.

Because themes can enqueue webfonts registered through `theme.json`, and they do that when hooking into `after_setup_theme` inside their `functions.php` file, running this hook on `wp_loaded` means it's too late to register the fonts.

I found that the default priority works just fine -- plugins are registered before themes. But let me know if this should change.

cc @aristath

Reply:
And now the tests are failing 😕  Any tips on how to fix it?

Reply:
Well... Thinking more about it.

Maybe we do not need this change.

Registration of fonts from `theme.json` can happen in `wp_loaded` and programmatically registration of fonts can happen in `after_setup_theme`, that's just fine.

We might need to change the documentation on `wp_enqueue_webfont` so that it's clear that manual enqueueing needs to happen after `wp_loaded` at least, so it's guaranteed that the fonts were registered.

Reply:
Also -- why is `gutenberg_register_webfonts_from_theme_json` hooked into `wp_loaded` and not into `init`, @aristath?